server/store/oauth2.go

```diff
@@ -71,7 +73,12 @@ func (s *oauth2Store) SaveUser(appID apps.AppID, actingUserID string, data []byt
 		return err
 	}
 
-	_, err = s.conf.MattermostAPI().KV.Set(userkey, data)
+	dataEncrypted, err := s.encrypter.Encrypt(string(data))
+	if err != nil {
+		return err
+	}
```

Comment:
Let's wrap the error for more context:
```suggestion
	if err != nil {
		return errors.Wrap(err, "failed to encrypt oauth user data")
	}
```